src/services/document_handler.py

```diff
@@ -0,0 +1,76 @@
+from fastapi import HTTPException
+from sqlalchemy import select, update
+from sqlalchemy.orm import Session
+from src.services.documents_utils import delete_file_from_s3, download_file_from_s3, upload_file_to_s3
+from src.services.project_manager_tables import Documents, Projects
+from src.routers.documents.schemas import Document
+from datetime import datetime
+
+class DocumentHandler():
+    @staticmethod
+    def download_document(document_id: int, db: Session):
+        doc = db.get(Documents, document_id)
+        key = doc.s3_key
+        name = doc.name
+        content_type = doc.content_type
+        contents = download_file_from_s3(bucket_name="project-manager-documents",
+                                         key=key)
+        return (name, content_type, contents)
+    
+
+    @staticmethod
+    def update_document(document_id: int,
+                        doc_name: str,
+                        content_type: str,
+                        updating_user: str,
+                        b_content: bytes,
+                        db: Session):
+        # create dict of attributes to update
+        fields_to_update = {"name": doc_name.strip().replace(" ", "-"),
+                            "added_by": updating_user,
+                            "content_type": content_type,
+                            "added_on": datetime.now()}
+        # get s3_key from db
+        key = db.get(Documents, document_id).s3_key
+        try:
+            q = update(Documents).where(Documents.id == document_id).values(
+                fields_to_update)
+            db.execute(q)
+            # try uploading new doc with old key to s3
+            upload_file_to_s3(bucket_name="project-manager-documents",

```

Comment:
About the code structure:
It is a good practice to create a separate class for any integrational thing. Like integration with DB, integration with S3 or any other AWS service, integration with third-party API
For example, for S3 you can have separate class
```
class S3Service:
    def __init__(bucket_name: str):
       self.bucket_name = bucket_name
       self.s3_client = ....
       self.s3_resource = ....
```       
So everyth would be in one place, you can pass the object of this class to __init__ of ProjectHandler and use one S3 client the whole time. Additionally, you can handle S3 related exception in this S3Service class (if key is not exists and others)